evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -0,0 +1,189 @@
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from evalml.pipelines import TimeSeriesRegressionPipeline
+from evalml.pipelines.components.transformers.transformer import Transformer
+
+
+@pytest.fixture
+def ts_data():
+    X, y = pd.DataFrame({"features": range(101, 132)}), pd.Series(range(1, 32))
+    y.index = pd.date_range("2020-10-01", "2020-10-31")
+    X.index = pd.date_range("2020-10-01", "2020-10-31")
+    return X, y
+
+
+class MockDelayedFeatures(Transformer):
```

Comment:
A lot of the complexity from time series comes from the nans that are introduced into the features. We need a component that does this to make the tests more robust. I'll delete this in favor of the bona-fide DelayedFeaturesTransformer when #1379 is merged.